src/components/DatatableV2/toolbar/Pagination.tsx

```diff
@@ -2,61 +2,88 @@ import React from 'react';
 
 import { DatatableInstance } from '../Datatable.types';
 
-export function useDebounce(value, delay) {
-  const [debouncedValue, setDebouncedValue] = React.useState(value);
-
-  React.useEffect(() => {
-    const handler = setTimeout(() => {
-      setDebouncedValue(value);
-    }, delay);
-
-    return () => {
-      clearTimeout(handler);
-    };
-  }, [value, delay]);
-
-  return debouncedValue;
-}
-
 const Pagination = <D,>({ table }: { table: DatatableInstance<D> }) => {
-  const currentPage = table.getState().pagination.pageIndex + 1;
-  const lastPage = table.getPageCount() - 1;
+  const {
+    getCanNextPage,
+    getCanPreviousPage,
+    getPageCount,
+    getPrePaginationRowModel,
+    getState,
+    nextPage,
+    options,
+    previousPage,
+    setPageIndex,
+    setPageSize,
+  } = table;
+  const { pagination } = getState();
+  const { pageIndex, pageSize } = pagination;
+
+  const currentPage = pageIndex + 1;
+  const lastPage = getPageCount() - 1;
+  const totalRowCount =
+    options.rowsCount ?? getPrePaginationRowModel().rows.length;
+  const firstRowIndex = pageIndex * pageSize;
+  const lastRowIndex = Math.min(pageIndex * pageSize + pageSize, totalRowCount);
```

Comment:
shouldn't this be ... totalRowCount -1?

Reply:
Nope this is used to display rows range on current page. E.g. I'm on page 2 and I display 20 rows per page so the firstRowIndex will be `21` and the lastRowIndex should be `40`